packages/driver/src/cy/net-stubbing/events/before-request.ts

```diff
@@ -119,8 +119,60 @@ export const onBeforeRequest: HandlerFn<CyHttpMessages.IncomingRequest> = (Cypre
   let resolved = false
   let handlerCompleted = false
 
+  const createQueryObject = () => {
+    try {
+      const url = new URL(req.url)
+      const urlSearchParams = new URLSearchParams(url.search)
+      const result = {}
+
+      for (let pair of urlSearchParams.entries()) {
+        result[pair[0]] = pair[1]
+      }
+
+      return result
+    } catch { // invalid url
+      return {}
```

Comment:
why not throw?

Reply:
Sometimes, `url` is `""` in [some test cases like these](https://app.circleci.com/pipelines/github/cypress-io/cypress/21304/workflows/b904acb0-cca0-465d-8383-bfbf742ce5b5/jobs/776989/parallel-runs/0?filterBy=FAILED). 

This is written in this way to avoid those cases.

Reply:
It's valid for the user to set `req.url` to an empty string or a relative path, it will be normalized against the last `req.url` on the backend. Can you add a test that this works with `req.url = '/some-path?a=b'`?